src/config/backends/splash-image.ts

```diff
@@ -0,0 +1,196 @@
+			buffer[7] === 0x0a
+		);
+	}
+
+	// Get the file path for the splash image for the underlying
+	// system
+	private async getSplashPath(): Promise<string> {
+		const [
+			// If no logo is defined, assume the file is `balena-logo.png`
+			splashFile = path.join(SplashImage.BASEPATH, 'balena-logo.png'),
+			// Read the splash dir (will throw if the path does not exist)
+		] = await Bluebird.resolve(fs.readdir(SplashImage.BASEPATH))
+			// Check for files in the valid splash filename list
+			.filter((filename) => SplashImage.FILENAMES.includes(filename))
+			// Convert to full path
+			.map((filename) => path.join(SplashImage.BASEPATH, filename));
+
+		return splashFile;
+	}
+
+	// Returns the base64 contents of the splash image
+	private async readSplashImage(where?: string): Promise<string> {
+		// Read from defaultPath unless where is defined
+		where = where ?? (await this.getSplashPath());
+
+		// read the image file...
+		return (await fs.readFile(where)).toString('base64');
+	}
+
+	// Write a splash image provided as a base64 string
+	private async writeSplashImage(image: string, where?: string): Promise<void> {
+		// Read from defaultPath unless where is defined
+		where = where ?? (await this.getSplashPath());
+
+		const buffer = Buffer.from(image, 'base64');
+		if (this.isPng(buffer)) {
+			// Write the buffer to the given location
+			await remountAndWriteAtomic(where, buffer);
+		} else {
+			throw new Error('Splash image should be a base64 encoded PNG image');
+		}
+	}
+
+	private stripPrefix(name: string): string {
+		if (!name.startsWith(SplashImage.PREFIX)) {
+			return name;
+		}
+		return name.substr(SplashImage.PREFIX.length);
+	}
+
+	public createConfigVarName(name: string): string {
+		return `${SplashImage.PREFIX}${name}`;
+	}
+
+	public async getBootConfig(): Promise<ConfigOptions> {
+		try {
+			const defaultImg = await this.readSplashImage(SplashImage.DEFAULT);
+			const img = await this.readSplashImage();
+
+			// If the image is the same as the default image
+			// return nothing
+			if (img !== defaultImg) {
+				return {
+					image: `data:image/png;base64,${img}`,
+				};
+			}
+		} catch (e) {
+			log.warn('Failed to read splash image:', e);
+		}
+		return {};
+	}
+
+	public async initialise(): Promise<SplashImage> {
+		try {
+			await super.initialise();
+
+			// The default boot image file has already
+			// been created
+			if (await fs.exists(SplashImage.DEFAULT)) {
+				return this;
+			}
+
+			// read the existing image file...
+			const image = await this.readSplashImage();
+
+			// write the image to the DEFAULT path
+			await this.writeSplashImage(image, SplashImage.DEFAULT);
+
+			log.success('Initialised splash image backend');
+		} catch (error) {
+			log.warn('Could not initialise splash image backend', error);
+		}
+		return this;
+	}
+
+	public ensureRequiredConfig(_deviceType: string, conf: ConfigOptions) {
+		// If the value from the cloud is empty, it is the same as no definition
```

Comment:
This is actually a very useful thing to have in the backends! I might use this for the ODMDATA backend when there's no mode specified.